libs/ledgerjs/packages/devices/src/ble/receiveAPDU.ts

```diff
@@ -65,11 +94,13 @@ export const receiveAPDU = (rawStream: Observable<Buffer>): Observable<Buffer> =
         if (notifiedData.length === notifiedDataLength) {
           o.next(notifiedData);
           o.complete();
-          sub.unsubscribe();
+          // Tries to unsubscribe from the raw stream as soon as we complete the parent observer
+          subscriptionCleaner.next();
```

Comment:
what's the advantage of doing this over `sub.unsubscribe()`?

Reply:
Good question: we're inside the definition of `next`, and we could do:
```
const subscription = rawStream.subscribe({
	next: () => {
		...
		subscription.unsubscribe();
   }
})
```

But at the moment of the `next` definition, `subscription` does not yet exist (and i TS might tell you you are using a variable not yet defined).

Reply:
it was working prior to this change though no ?

Reply:
but yeah agreed it was weird